### DISCUSSION TITLE
sCF incorrectly mapping values on target tree

### DISCUSSION THREAD

**Discussion by veronicamb** [iqtree/iqtree2#306] [Q&A] [answered]
Hello all, 
I'm trying to run a site Concordance Factor (sCF) analysis but I'm running into a strange issue. I noticed that, when mapping the sCF values onto the target tree, it appears that the node labels somehow moved, and values were mapped to incorrect nodes.

The target tree already has 2 labels on it: ufboot and sh-alrt, so sCF should is appended to them. That's how I noticed the inconsistencies in node labels.

I checked the .branch files, and the node_IDs seem to be correct. However, looking at the .stat files I noticed a discrepancy between the sCF.stat and the gCF.stat files (attached below). gCF was mapped correctly onto the target tree. In the stats files, the first few nodes are correct and then they appear to go "out of sync". 

[gCF.cf.stat.txt](https://github.com/user-attachments/files/16733898/gCF.cf.stat.txt)
[sCF.cf.stat.txt](https://github.com/user-attachments/files/16733900/sCF.cf.stat.txt)

This is the command I used to run sCF:

`iqtree -te ../target.treefile -spp ../sCF.best_scheme.nex --scfl 100 -p ../${LOCI} -pre sCF -nt AUTO -ntmax 5`

Also, I have a second question! I tried to give sCF a model so as to skip modeltest, but I'm clearly doing it wrong because it doesn't read the file .best_scheme.nex and runs ModelTest instead. Is there another flag I should be using instead of -spp?

thank you very much!

Veronica

**Comment by roblanf** [comment]
Hi Veronica,

Thanks for bringing this to our attention! 

Could you post all of your commands, the labelled output trees, the .iqtree and .log files, and (if possible) the input alignments too?

Thanks,

Rob

**Comment by veronicamb** [comment, marked as answer]
Hello Rob, 

It seems that I have found a solution to this issue, although I don't quite understand why. 

In the analysis I ran before, I actually used a slightly different command from what I posted above:

`iqtree -t ../target.treefile -spp ../sCF.best_scheme.nex --scfl 100 -p ../${LOCI} -pre sCF -nt AUTO -ntmax 5`

I ran the analysis again using the command I posted above:

`iqtree -te ../target.treefile -spp ../sCF.best_scheme.nex --scfl 100 -p ../${LOCI} -pre sCF -nt AUTO -ntmax 5`

And that worked perfectly. 

Apparently what I did wrong was that instead of `-te target.treefile`, I used `-t target.treefile` and that got me that tree with incorrectly mapped support.

I guess my question is: is there a difference between `-te` and `-t`?

I'd be happy to share further files, but if that's okay, I'd be more comfortable sharing them privately, because the data belongs to my collaborators on this study.

Here's the .stat file for the second run that mapped support correctly.

[NT_Elasmo_PTT_sCF.cf.stat.txt](https://github.com/user-attachments/files/16749239/NT_Elasmo_PTT_sCF.cf.stat.txt)

Here are the .iqtree and .log files for both runs (`-t `and `-te`)


[Command_using_t_sCF.iqtree.txt](https://github.com/user-attachments/files/16749332/Command_using_t_sCF.iqtree.txt)
[Command_using_t_sCF.log.txt](https://github.com/user-attachments/files/16749333/Command_using_t_sCF.log.txt)
[Command_using_te_sCF.iqtree.txt](https://github.com/user-attachments/files/16749334/Command_using_te_sCF.iqtree.txt)
[Command_using_te_sCF.log.txt](https://github.com/user-attachments/files/16749341/Command_using_te_sCF.log.txt)

Thank you for your patience!! 

Veronica

**Reply by roblanf** [reply]
Hi Veronica,

This makes perfect sense. When you use `-t` you are supplying a starting tree, and then re-optimising that tree. I.e. your branch IDs are for two (probably different) trees. When you use `-te` you are fixing the tree for the whole analysis, so the branch IDs should match up.

Hopefully that helps!

Rob

**Comment by veronicamb** [comment]
Hi Rob, 

Now everything makes so much more sense, thank you! That also explains why the branch lengths were a bit off.

Thank you for the clarification!

best,

Veronica
